unit-tests/src/testing_utils/neutron_adapters/stargate_module.rs

```diff
@@ -0,0 +1,126 @@
+use cosmos_sdk_proto::traits::MessageExt;
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{
+    coins, from_json, to_json_binary, Addr, AnyMsg, Api, Binary, BlockInfo, Coin, CustomMsg,
+    CustomQuery, GrpcQuery, Querier, Storage, Uint64,
+};
+use cw_multi_test::{
+    error::{anyhow, AnyResult},
+    AppResponse, CosmosRouter, Stargate,
+};
+use neutron_sdk::proto_types::neutron::interchaintxs::v1::{Params, QueryParamsResponse};
+
+use serde::de::DeserializeOwned;
+
+pub struct StargateModule;
+
+impl Stargate for StargateModule {
+    fn execute_stargate<ExecC, QueryC>(
+        &self,
+        _api: &dyn Api,
+        _storage: &mut dyn Storage,
+        _router: &dyn CosmosRouter<ExecC = ExecC, QueryC = QueryC>,
+        _block: &BlockInfo,
+        sender: Addr,
+        type_url: String,
+        value: Binary,
+    ) -> AnyResult<AppResponse>
+    where
+        ExecC: CustomMsg + DeserializeOwned + 'static,
+        QueryC: CustomQuery + DeserializeOwned + 'static,
+    {
+        cw_multi_test::error::bail!(
+            "Unexpected stargate execute in custom StargateModule impl: type_url={}, value={} from {}",
+            type_url,
+            value,
+            sender,
+        )
+    }
+
+    fn query_stargate(
+        &self,
+        _api: &dyn Api,
+        _storage: &dyn Storage,
+        _querier: &dyn Querier,
+        _block: &BlockInfo,
+        path: String,
+        _data: Binary,
+    ) -> AnyResult<Binary> {
+        println!("stargate query mock received for path: {path}");
+        if path == "/neutron.interchaintxs.v1.Query/Params" {
```

Comment:
Would probably do a match here so extending it will be much easier.

Reply:
yeah just couldn't think of what else was needed so ended up with this for now, totally gonna switch to match with the next addition